app/controllers/ontologies_controller.rb

```diff
@@ -437,6 +437,61 @@ def metrics_evolution
     render partial: 'ontologies/sections/metadata/metrics_evolution_graph', locals: { data: data }
   end
 
+  def selector
+    @categories = LinkedData::Client::Models::Category.all(display_links: false, display_context: false)
+    @groups = LinkedData::Client::Models::Group.all(display_links: false, display_context: false)
+    @filters = ontology_filters_init(@categories, @groups)
+    @select_id = params[:id]
+    render 'ontologies/selector/selector' , layout: false
+  end
+
+  def selector_results
+    @ontologies = LinkedData::Client::Models::Ontology.all(include_views: params[:showOntologyViews])
+    @total_ontologies_number = @ontologies.length
+    @input = params[:input] || ''
+    @ontologies = @ontologies.select { |ontology| ontology.name.downcase.include?(@input.downcase) || ontology.acronym.downcase.include?(@input.downcase)}
+
+    if params[:groups] 
+      @ontologies = @ontologies.select do |ontology|
+        (ontology.group & params[:groups]).any?
+      end
+    end
+
+    if params[:categories]
+      @ontologies = @ontologies.select do |ontology|
+        (ontology.hasDomain & params[:categories]).any?
+      end
+    end
+
+    if params[:formats] || params[:naturalLanguage] || params[:formalityLevel] || params[:isOfType] || params[:showRetiredOntologies]
+      submissions = Rails.cache.fetch('all_submissions') do
```

Comment:
if you use cache here, it needs to be invalidated when a submission or any child object gets updated. (e.g. an ontology changes its name or status, ....) otherwise, this value will never change.